color/src/serialize.rs

```diff
@@ -77,19 +77,69 @@ fn write_legacy_function(
-            ColorSpaceTag::Hsl => write_legacy_function(self, "hsl", 1.0, f),
-            ColorSpaceTag::Hwb => write_modern_function(self, "hwb", f),
-            ColorSpaceTag::XyzD65 => write_color_function(self, "xyz", f),
-            ColorSpaceTag::Lab => write_modern_function(self, "lab", f),
-            ColorSpaceTag::Lch => write_modern_function(self, "lch", f),
-            ColorSpaceTag::Oklab => write_modern_function(self, "oklab", f),
-            ColorSpaceTag::Oklch => write_modern_function(self, "oklch", f),
+        if self.flags.named() {
+            if let Some(color_name) = self.flags.color_name() {
+                return write!(f, "{}", color_name);
+            }
+
+            match self.cs {
+                ColorSpaceTag::Srgb => write_legacy_function(self, "rgb", 255.0, f),
+                ColorSpaceTag::Hsl | ColorSpaceTag::Hwb => {
+                    let srgb = self.convert(ColorSpaceTag::Srgb);
+                    write_legacy_function(&srgb, "rgb", 255.0, f)
+                }
+                ColorSpaceTag::Lab => write_modern_function(self, "lab", f),
+                ColorSpaceTag::Lch => write_modern_function(self, "lch", f),
+                ColorSpaceTag::Oklab => write_modern_function(self, "oklab", f),
+                ColorSpaceTag::Oklch => write_modern_function(self, "oklch", f),
+                _ => unreachable!(),
+            }
+        } else {
+            let color_space = match self.cs {
+                ColorSpaceTag::Srgb => "srgb",
+                ColorSpaceTag::LinearSrgb => "srgb-linear",
+                ColorSpaceTag::DisplayP3 => "display-p3",
+                ColorSpaceTag::Hsl => "hsl",
+                ColorSpaceTag::Hwb => "hwb",
+                ColorSpaceTag::XyzD65 => "xyz",
+                ColorSpaceTag::Lab => "lab",
+                ColorSpaceTag::Lch => "lch",
+                ColorSpaceTag::Oklab => "oklab",
+                ColorSpaceTag::Oklch => "oklch",
+            };
+            write_color_function(self, color_space, f)
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::parse_color;
+
+    #[test]
+    fn specified_to_serialized() {
+        for (specified, expected) in [
+            ("rgb(1,1,1)", "rgb(1, 1, 1)"),
+            // TODO: output rounding? Otherwise the tests should check for approximate equality
+            // (and not string equality) for these conversion cases
+            (
+                "hwb(740deg 20% 30% / 50%)",
+                "rgba(178.5, 93.50008, 50.999996, 0.5)",
+            ),
+            // currently fails, but should succeed (values should be clamped at parse-time)
+            // ("rgb(1.1,1,1)", "rgb(1, 1, 1)"),
+            ("color(srgb 1.0 1.0 1.0)", "color(srgb 1 1 1)"),
+            ("rosybrown", "rosybrown"),
+            ("red", "red"),
+            ("transparent", "transparent"),
+            ("yellowgreen", "yellowgreen"),
```

Comment:
There's a simple exhaustive test that the named colours all roundtrip. I think it could only go wrong if the phf were wrong (?), but that would still be good to catch. I don't think that test would take *too* long to run.